### DISCUSSION TITLE
Allow graphql queries to orderBy subfields

### DISCUSSION THREAD

**Discussion by riekusdn** [craftcms/cms#16157] [Ideas] [answered]
It would be very helpful if the `orderBy` in graphql queries would support the use of subfields. As far as I can tell from the documentation and my own experience, you can only order by a field or multiple fields, but not by a sub-field of, say, a Neo or Matrix field. 

My use case: I'm using https://github.com/studioespresso/craft-date-range for an events related entry type. The `dateRange` field consists of a `start` and `end` sub-field. It seems impossible to use `dateRange.end` or `dateRange.start` in my `orderBy`.

**Comment by brandonkelly** [comment, marked as answer]
Just made this possible for Craft 5.6 (a6c770d543818434eff6e672c12309a80d1c1e59).

If you’d like to test it out, change your `craftcms/cms` requirement in composer.json to:

```json
"craftcms/cms": "5.6.x-dev as 5.6.0-alpha",
```

and then run `composer update`.

Alternatively, you could create new [Preparse fields](https://plugins.craftcms.com/preparse-field?craft5) that extract the `start` and `end` values into their own (hidden) custom fields, and order by those fields instead.

**Reply by brandonkelly** [reply]
Craft 5.6.0 is out now 🚀
